Core/Src/main.c

```diff
@@ -92,40 +87,53 @@ void HAL_UARTEx_RxEventCallback(UART_HandleTypeDef *huart, uint16_t Size)
 /**
  * XOR logic used to divide data by key
  */
-void xor(){
-	if(bit_ans & 0b1000){			//if leftmost bit is 1, perform xor with key
-		bit_ans = bit_ans ^ bit_key;
-	}else{						//if leftmost bit is 0, perform xor with all zeros
-		bit_ans = bit_ans ^ 0b0000;
+uint64_t crc_xor(uint64_t div_data){
```

Comment:
Is 64-bit required, or can this be optimized for 32-bit variables? 64-bit will operations will run much slower. Not sure if the execution speed will be an issue though.

Reply:
I made them 64bit since ideally the code can track up to 32 button states (exactly 32 bits). To calculate the crc value, the 32bit data must be shifted and 3 zeros have to be appended to the end (therefore taking 35 bits). If we only tracked 29 buttons for example, I would be able to keep it uint32 variables

Reply:
Makes sense. Good to know -- we can probably drop to 32-bit and 29 buttons if needing to speed things up.